qiskit/circuit/library/basis_change/qft.py

```diff
@@ -103,6 +104,8 @@ def __init__(
         self._insert_barriers = insert_barriers
         self._inverse = inverse
         self.num_qubits = num_qubits
+        # Warn immediately if we can detect straight away that there is a problem.
+        self._warn_if_precision_loss()
```

Comment:
I see why you put this here, though it's a little inconsistent with not warning on the qubit setter, since I could just do
```python
qft = QFT(2000)
qft.num_qubits = 1
# and only now use it
```
but that's not a very sensible use case and if you want to keep it here I'm good.

Reply:
Yeah, you're right about inconsistency, actually - it'll also warn twice if you instantiate then build without changing anything.

I'm prepared to go with whichever way you think is better.  I originally only had the warning in the initialisation, because I forgot that your algorithms stuff allows delayed mutation of the number of qubits and approximation degree - that's a very unnatural workflow for me, so I don't know what you would find most helpful.  I didn't want to warn on set of `num_qubits` or `approximation_degree` individually, because setting them both is necessarily a non-atomic operation, and the state may become valid after _both_ are changed.

The disadvantage to _only_ warning in `_build` is that you don't get the warning until you've just embarked on a calculation that will be rather slow - I generally prefer to eagerly warn on creation of a bad state, but I also generally programme with immutability where possible, which isn't the same style you use here.

Reply:
In my opinion it would be a bit better to just keep it in `_check_configuration` to stay consistent with the mutability of blueprint circuits. 🙂

Reply:
Fine by me.  Done in 10f76b1.